universal-application-tool-0.0.1/app/services/program/ProgramServiceImpl.java

```diff
@@ -394,12 +393,9 @@ public void publishProgram(long id) throws ProgramNotFoundException {
 
   @Override
   public ProgramDefinition newDraftOf(long id) throws ProgramNotFoundException {
-    ProgramDefinition program =
-        this.getProgramDefinition(id).toBuilder().setLifecycleStage(LifecycleStage.DRAFT).build();
-    Program programBean = program.toProgram();
-    programRepository.insertProgramSync(programBean);
-    programRepository.updateQuestionVersions(programBean);
-    return programBean.getProgramDefinition();
+    return programRepository
+        .updateOrCreateDraft(this.getProgramDefinition(id).toProgram())
```

Comment:
Nit (non-blocking): from this context, I think create draft is more common, so `createOrUpdateDraft` is more readable to me. Your call to update it or not.

Reply:
Agree, done.